src/components/panel/panel.spec.js

```diff
@@ -1880,6 +1880,31 @@ describe('$mdPanel', function() {
         });
       });
 
+      it('should have assigned the actual position by the time the offset' +
+        'methods have been called', function() {
+          var hasActualPosition = false;
+
+          var obj = {
+            getOffsetX: function(position) {
```

Comment:
Consider making this entire method a spy and then doing something like this:

```
var getOffsetX = jasmine.createSpy('position');
var position = mdPanelPosition
    .relativeTo(myButton[0])
    .addPanelPosition(xPosition.ALIGN_START, yPosition.ALIGN_TOPS)
    .withOffsetX(getOffsetX);

config['position'] = position;

openPanel(config);

var positionObj = jasmine.objectContaining({
  _actualPosition: {
    x: xPosition.ALIGN_START,
    y: yPosition.ALIGN_TOPS
  }
};

expect(getOffsetY).toHaveBeenCalledWith(positionObj);
```

An even better way is to write this test using `getActualPosition` since that's actually the method we can access. Typescript makes private stuff really private, so you can't even write tests with them.